src/main/webapp/script.js

```diff
@@ -38,12 +38,123 @@ function isSignedIn() {
   });
 }
 
-// Show the sign in button.
-function displaySignInButton() {
-  document.getElementById('sign-in-button-form').style.display = 'block';
+// Create the necessary elements and display the sign in page.
+function displaySignInPage() {
+  // Set the width of the content container.
+  setContentWidth('650px');
+
+  // Add background image to site. (This id is defined in CSS.)
+  document.body.id = 'body-background-image';
+
+  // Get the homepage "registration" block to and add elements.
+  const indexRegistrationBlock = document.getElementById('index-registration-block');
+  indexRegistrationBlock.style.display = 'block';
+
+  // Create h1 element as title for TravIS.
+  const titleElement = document.createElement('h1');
+  titleElement.innerText = 'TravIS: Travel Information System';
+  titleElement.className = 'index-registration-block-child';
+
+  // Create p element to describe TravIS.
+  const textElement = document.createElement('p');
+  textElement.innerText = 
+    'TravIS, your personal Travel Information System, allows you to plan ' +
+    'the optimal trip to wherever you want to visit.';
+  textElement.className = 'index-registration-block-child';
+  textElement.id = 'index-registration-block-p';
+
+  // Create form element to allow sign in.
+  const formElement = document.createElement('form');
+  formElement.className = 'index-registration-block-child';
+  formElement.id = 'sign-in-button-form';
+  formElement.action = '/auth';
+  formElement.method = 'POST';
+
+  // Create input element to allow sign in within the form element.
+  const inputElement = document.createElement('input');
+  inputElement.className = 'btn btn-primary';
+  inputElement.type = 'submit';
+  inputElement.value = 'Sign in';
+
+  // Add input element to the form element.
+  formElement.appendChild(inputElement);
+
+  // Add above elements to the homepage "registration" block.
+  indexRegistrationBlock.appendChild(titleElement);
+  indexRegistrationBlock.appendChild(textElement);
+  indexRegistrationBlock.appendChild(formElement);
+}
+
+// User is signed in, show the start trip homepage.
+function displayStartTripDesign() {
+  // Set the width of the content container.
+  setContentWidth('800px');
+
+  // Remove background image id from body.
+  document.body.removeAttribute('id');
+
+  // Get the homepage "start trip" block to and add elements.
+  const indexStartTripBlock = document.getElementById('index-start-trip-block');
+  indexStartTripBlock.style.display = 'block';
```

Comment:
If I start the page an click sign-in then the trips button I see the following error in the js console coming from this line:
script.js:26 TypeError: Cannot read property 'style' of null
    at displayStartTripDesign (script.js:98)
    at script.js:20

I think because clicking on the trips button redirects to /trips the 'index-start-trip-block' element no longer exists when this method is being called. But I'm not certain.

Reply:
Yes, I was aware of that and I fixed it in my next CL -- I'll try to remove this issue from this CL as well.

Reply:
Fixed!